app/services/foreman/access_control.rb

```diff
@@ -29,6 +29,21 @@ def permissions
         @permissions
       end
 
+      def permissions_for_controller_action(controller_action)
+        controller_action = path_hash_to_string(controller_action) if controller_action.is_a?(Hash)
+        @permissions.select { |p| p.actions.include?(controller_action) }
+      end
+
+      def normalize_path_hash(hash)
+        hash[:controller] = hash[:controller].to_s.gsub(/::/, "_").underscore
```

Comment:
`.tr` won't work here, it would make it double underscore as it replaces one char at a time